src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -597,6 +599,10 @@ private CallResult ExecuteCall(EvmState vmState, byte[]? previousCallResult, Zer
     {
         bool isTrace = _logger.IsTrace;
         bool traceOpcodes = _txTracer.IsTracingInstructions;
+        (bool IsOn, DebugTracer Debugger) debugMode = _txTracer is DebugTracer dbgTracer
```

Comment:
maybe remove `IsOn` and only rely on `Debugger` - btw those are local variables, should be `isOn` and `debugger`